GliaWidgets/ViewModel/EngagementViewModel.swift

```diff
@@ -115,10 +115,14 @@ class EngagementViewModel {
         update(for: interactor.state)
     }
 
-    func enqueue() {
-        interactor.enqueueForEngagement {} failure: { error in
-            self.handleError(error)
-        }
+    func enqueue(mediaType: MediaType) {
+        interactor.enqueueForEngagement(
+            mediaType: mediaType,
+            success: {},
+            failure: { error in
+                self.handleError(error)
```

Comment:
since we pass self into closure, let's avoid memory leaking and use weak self here.

Reply:
good catch 😄